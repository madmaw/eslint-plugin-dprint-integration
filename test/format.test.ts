import { readFileSync, readdirSync } from "node:fs";

import { Linter } from "eslint";
import * as parserPlain from "eslint-parser-plain";
import { describe, expect, it } from "vitest";

import DprintIntegration from "../src";

const linter = new Linter();
const linterConfig = {
	parser: "plain",
	rules: {
		dprint: [
			"error",
			{},
			{
				malva: {
					declarationOrder: "alphabetical",
				},
			},
		],
	},
} satisfies Linter.Config;

linter.defineParser("plain", parserPlain as any);
linter.defineRule("dprint", DprintIntegration.rules.dprint);

describe("format", () => {
	const folder = "./test/__fixtures__/format/";
	const files = readdirSync(folder);
	const fixtureFiles = files.map<[string, string]>((file) => [
		file,
		readFileSync(folder + file, "utf-8"),
	]);

	const results = fixtureFiles.map(([file, source]) => {
		const { output, messages, fixed } = linter.verifyAndFix(
			source,
			linterConfig,
			file,
		);

		return {
			file,
			output,
			messages,
			fixed,
		};
	});

	for (const { file, output, messages, fixed } of results) {
		it(`File: ${file}`, async () => {
			expect(output).toMatchSnapshot();
			expect(messages).toMatchSnapshot();
			expect(fixed).toBeTruthy();
		});
	}
});
