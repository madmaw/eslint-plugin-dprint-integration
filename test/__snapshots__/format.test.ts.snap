// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`format > File: css.css 1`] = `
".class {
  background-color: #fff;
  font-family: "Courier New";
  height: 200px;
  width: 100px;
}
"
`;

exports[`format > File: css.css 2`] = `[]`;

exports[`format > File: svelte.svelte 1`] = `
"<div></div>
"
`;

exports[`format > File: svelte.svelte 2`] = `[]`;

exports[`format > File: typescript.ts 1`] = `
"const a = "1";
"
`;

exports[`format > File: typescript.ts 2`] = `[]`;

exports[`format > File: vue.vue 1`] = `
"<template>
  <div></div>
</template>
"
`;

exports[`format > File: vue.vue 2`] = `[]`;
